### PR TITLE
Feature/add course endpoint

### PR DESCRIPTION
**Pull request by aaronpear**
## Description

This pull request adds necessary role authentication middleware to the POST add course endpoint such that only instructors, admins, or super admins may access it and other users such as parents or their children may not. Middleware for parent and student authentication were built as well for future use.

## Loom Video

https://www.loom.com/share/3629ddd12bde4e19bc00264061b73fc3

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have removed unnecessary comments/console logs from my code
- [ ] I have made corresponding changes to the documentation if necessary (optional)
- [x] My changes generate no new warnings
- [x] I have checked my code and corrected any misspellings
- [x] No duplicate code left within changed files
- [x] Size of pull request kept to a minimum
- [x] Pull request description clearly describes changes made & motivations for said changes

### CODE REVIEW

File: api/middleware/roleAuthentication.js [hunk 1]
```diff
@@ -1,15 +1,43 @@
-const { getRoleById } = require('../roles/rolesModel');
```
Comment (jmerz826) [inline api/middleware/roleAuthentication.js:1]: Love this refactor- minimizes calls to DB.

File: api/middleware/roleAuthentication.js [hunk 2]
```diff
@@ -1,15 +1,43 @@
...
+/*
+  Roles:
+  1 - super_admin
+  2 - admin  
+  3 - instructor
+  4 - parent
+  5 - child
+*/
+// Only allows endpoint access for Super Admins (1), Admins (2), and Instructors (3)
+const roleAuthenticationInstructor = (req, res, next) => {
```
Comment (jmerz826) [inline api/middleware/roleAuthentication.js:10]: These 3 functions are okay for now, but would benefit from a refactor (for readability/DRY purposes).

**Review by jmerz826 — APPROVED** [review]

**Review by JakeHappel — APPROVED** [review]
Pretty good implementation for the roleauthentication and will be essential for many other endpoints.

**Review by ryan-hamblin — APPROVED** [review]
I've been excited about this Pull Request for a while. Getting us all onto the same page about our `Courses` is so important for this app. Great work here. Everything is nice and tidy.

**Pull request merged**